aya/src/bpf.rs

```diff
@@ -389,9 +389,26 @@ impl<'a> BpfLoader<'a> {
 
         let btf_fd = if let Some(features) = &FEATURES.btf() {
             if let Some(btf) = obj.fixup_and_sanitize_btf(features)? {
-                // load btf to the kernel
-                let btf = load_btf(btf.to_bytes(), *verifier_log_level)?;
-                Some(btf)
+                match load_btf(btf.to_bytes(), *verifier_log_level) {
+                    Ok(btf_fd) => Some(btf_fd),
+                    // Only report an error here if the BTF is truely needed, otherwise proceed without.
+                    Err(err) => {
+                        for program in obj.programs.values() {
+                            match program.section {
+                                ProgramSection::Extension { .. }
+                                | ProgramSection::FEntry { .. }
+                                | ProgramSection::FExit { .. } => {
+                                    return Err(BpfError::BtfError(err))
+                                }
+                                _ => {}
```

Comment:
I think instead of using `_` here, we should handle all the program sections explicitly, so that when we add new program types it'll fail to compile if this code isn't updated